File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/observer/lab/Manager.java
// Manager.java

public class Manager
{
	public Manager()
	{
		myMeetings = new DpVector<Meeting>();
	}

	public void add(Meeting aMeeting)
	{
		if (aMeeting == null)
		{
			throw new DpException("void Manager.add(Meeting aMeeting) -- Invalid aMeeting.");
		}

		System.out.println(
			  "Manager.add("
			+ aMeeting.toString()
			+ ")"
			);
		myMeetings.add(aMeeting);
	}

	public void remove(Meeting aMeeting)
	{
		if (aMeeting == null)
		{
			throw new DpException("void Manager.remove(Meeting aMeeting) -- Invalid aMeeting.");
		}

		System.out.println(
			  "Manager.remove("
			+ aMeeting.toString()
			+ ")"
			);
		// I was too lazy to implement this. :-(
	}

	public void update(Meeting aMeeting)
	{
		if (aMeeting == null)
		{
			throw new DpException("void Manager.update(Meeting aMeeting) -- Invalid aMeeting.");
		}

		System.out.println(
			  "Manager.update("
			+ aMeeting.toString()
			+ ")"
			);
	}

	private DpVector<Meeting> myMeetings;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/observer/lab/Manager.m
% Manager.m

classdef Manager < handle
	properties (GetAccess = private, SetAccess = private)
		Meetings % DpVector of Meeting
	end

	methods
		% constructor:
		function theManager = Manager
			theManager.Meetings = DpVector;
		end

		function add(theManager, aMeeting)
			if isempty(aMeeting)
				error('Invalid aMeeting.');
			end

			fprintf('Manager.add(%s)\n', aMeeting.toString);
			theManager.Meetings.add(aMeeting);
		end

		function remove(theManager, aMeeting)
			if isempty(aMeeting)
				error('Invalid aMeeting.');
			end

			fprintf('Manager.remove(%s)\n', aMeeting.toString);
			% I was too lazy to implement this. :-(
		end

		function update(theManager, aMeeting)
			if isempty(aMeeting)
				error('Invalid aMeeting.');
			end

			fprintf('Manager.update(%s)\n', aMeeting.toString);
		end
	end
end
